src/structures/ApplicationCommand.ts

```diff
@@ -0,0 +1,39 @@
+import BulbBotClient from "./BulbBotClient";
+import { CommandInteraction, Permissions, PermissionString } from "discord.js";
+import { ApplicationCommandOptions, ApplicationCommandType } from "../utils/types/ApplicationCommands";
+
+export default class ApplicationCommand {
+	public readonly client: BulbBotClient;
+	public readonly type: ApplicationCommandType;
+	public readonly name: string;
+	public readonly description: string;
+	public readonly dm_permissions: boolean;
+	public readonly default_member_permissions: string | null;
```

Comment:
Why `string | null`? Why don't we just store it as a `bigint`?

Reply:
Doesn't discord.js send us the the value as a string? cc @KlukCZ

Reply:
The Discord API sends the value as a string. My understanding has been that this was for compatibility reasons. I believe DJS provides it via a Permissions object, which supports various modes of access. One of them should be getting a bigint directly, if I'm not mistaken

Reply:
The API does send it as a `string` or `null` yeah. I haven't explored what options d.js gives us but if the Permissions object does work like wakfi described I don't have an issue switching to that